sapi/cli/php_http_parser.c

```diff
@@ -623,6 +624,8 @@ size_t php_http_parser_execute (php_http_parser *parser,
         } else if (parser->method == PHP_HTTP_MKCOL) {
           if (index == 1 && ch == 'O') {
             parser->method = PHP_HTTP_MOVE;
+          } else if (index == 3 && ch == 'A') {
+            parser->method = PHP_HTTP_MKCALENDAR;
```

Comment:
I can see that this code works, but I can't get my head around *why* it works. When control enters with `index=1` (and `ch=K`), I'd expect `parser->method` to be set to `NOT_IMPLEMENTED`, so that this branch is not entered again for a higher index.

What am I missing?